### PR TITLE
created_at for profiles, with a default of jan 26, 2016

### LINKED ISSUE #514: Add "created" field to profile and user model
Currently, we don't record a create date on new profiles or new users. This fall, we aim to onboard program applicants more diligently (e.g. awards, mozfest). Create date will help us track whether our work is effective or not. Let's add it.

### PR DESCRIPTION
**Pull request by Pomax**
Closes #514 by adding a `created_at` property that defaults to Jan 26th, 2016, if it wasn't set before.

### CODE REVIEW

**Comment by alanmoo** [comment]
@xmatthewx you wanted user accounts to have a creation date too, right?

**Comment by xmatthewx** [comment]
yes. it's useful to measure growth and could be useful if we need to triage some spam or abuse.

File: pulseapi/profiles/migrations/0022_userprofile_created_at.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+from django.utils.dateparse import parse_date
+from django.utils.timezone import make_aware
+
+def default_start_time():
+    date_str = "2016-01-26"
+    default_date = parse_date(date_str)
+    return make_aware(default_date)
```
Comment (gideonthomas) [inline pulseapi/profiles/migrations/0022_userprofile_created_at.py]: I think `default_date` needs to be a `datetime` object not a `date` object...that's probably why travis is failing as well.

You probably want to use something like `parse_datetime('2016-01-26 00:00:00')` as the default date instead and then you can make it timezone aware.

Comment (alanmoo) [reply pulseapi/profiles/migrations/0022_userprofile_created_at.py]: @Pomax Can you get this over the edge?

Comment (Pomax) [reply pulseapi/profiles/migrations/0022_userprofile_created_at.py]: yar

**Comment by Pomax** [comment]
I've updated this to be two migrations: one that adds the field, and makes it default to jan 1st, 2016, and a second migration which then switches the field definition over to always be whatever "now" is when the profile gets created.

This needed two migrations because `default` and `auto_now_add` are mutually exclusive,  and we wanted a migration that did both.

File: pulseapi/profiles/migrations/0022_userprofile_created_at.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    dependencies = [
+        ('profiles', '0021_auto_20190508_1858'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='userprofile',
+            name='created_at',
+            field=models.DateTimeField(default=datetime.datetime(2016, 1, 26, 0, 0, tzinfo=utc)),
```
Comment (Pomax) [inline pulseapi/profiles/migrations/0022_userprofile_created_at.py]: generated by `makemigration` based on:
```
    date_str = '2016-01-26 00:00:00'
    default_date = parse_datetime(date_str)
    timezoned = make_aware(default_date)
    created_at = models.DateTimeField(
        default=timezoned
    )
```

**Comment by xmatthewx** [comment]
Good catch. I don't think it matters though.

**Comment by Pomax** [comment]
Kept the field in the admin, but added it to the exclusion list for the public API

File: pulseapi/profiles/migrations/0023_auto_20190807_1733.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    dependencies = [
+        ('profiles', '0022_userprofile_created_at'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='userprofile',
+            name='created_at',
+            field=models.DateTimeField(auto_now_add=True),
```
Comment (Pomax) [inline pulseapi/profiles/migrations/0023_auto_20190807_1733.py]: immediate second migration that changes the identity of `created_at` to one that can only ever be "whatever the date/time is right now" when a profile gets created.

**Pull request merged**